File: exams/1617/translate/layout/solution/layout/fixed-layout.cpp
#include "fixed-layout.h"


class FixedLayout : public Layout
{
	Box m_box;

public:
	FixedLayout(const Box& box)
		: m_box(box) { }

	Box box() const override
	{
		return m_box;
	}
};

std::shared_ptr<Layout> create_fixed_layout(const Box& box)
{
	return std::make_shared<FixedLayout>(box);
}

File: exams/1617/translate/layout/student/java/FixedLayout.java
public class FixedLayout implements Layout
{
    private Box box;

    public FixedLayout(Box box)
    {
        this.box = new Box(box);
    }

    public Box box()
    {
        return new Box(box);
    }

    // Te vertalen als globale functie, d.i. een functie buiten de klasse
    public static Layout createFixedLayout(Box box)
    {
        return new FixedLayout(box);
    }
}
